aea/connections/local/connection.py

```diff
@@ -322,14 +323,17 @@ class OEFLocalConnection(Connection):
     It is useful for local testing.
     """
 
-    def __init__(self, public_key: str, local_node: LocalNode, connection_id: str = "local"):
+    supported_protocols = {"default", "oef", "fipa", "tac"}
```

Comment:
I feel this approach has a problem of "code scalability": let's say we have hundreds of protocols, and that the connections can support them. We should update constantly this variable (the same holds for its default configuration file).

Reply:
Yes good point. An alternative could be to support by default any protocol (i.e. when `{}` is passed to supported_protocols) and then only if supported_protocols is specified different from `{}` will the filtering kick in.

Reply:
Maybe this is the right approach anyway, so relabel `supported_protocols` to `restricted_to_protocols` and this is an optional param. If I pass nothing then no filtering takes place.